src/main/webapp/view/transcript/transcript.js

```diff
@@ -113,60 +85,106 @@ function seekTranscript(currentTime) {
   if (currentTimeMs < currentTranscriptLine.startTimestampMs) {
     return;
   }
-  if (isWithinCurrentTimeRange(currentTimeMs)) {
-    addBold(currentTranscriptLine);
+  if (currentTranscriptLine.isWithinCurrentTimeRange(currentTimeMs)) {
+    currentTranscriptLine.addBold();
     return;
   }
-  removeBold(currentTranscriptLine);
+  currentTranscriptLine.removeBold();
   currentTranscriptLine = currentTranscriptLine.nextElementSibling;
-  scrollToTopOfTranscript(currentTranscriptLine);
-  addBold(currentTranscriptLine);
+  currentTranscriptLine.scrollToTopOfTranscript();
+  currentTranscriptLine.addBold();
   // TODO: Handle the case where the video isn't only playing.
 }
 
 /**
- * Bolds the text in `transcriptLineLiElement` if it is not already
- * bolded.
+ * Creates a transcript line element containing the text,
+ * start time, and end time from `transcriptLine`.
  */
-function addBold(transcriptLineLiElement) {
-  if (isBolded(transcriptLineLiElement)) {
-    return;
+class TranscriptLine extends HTMLElement {
+  /**
+   * Creates a custom HTML element representing a transcript line.
+   *
+   * <p>Uses the template and slots defined in `TRANSCRIPT_TEMPLATE` to
+   * help create the transcript line.
+   *
+   * @param transcriptLine The transcriptLine from `ENDPOINT_TRANSCRIPT`
+   *     whose `attributes` should be used.
+   */
+  constructor(transcriptLine) {
+    super();
+    const timestampRange = window.createTimestampRange(
+        transcriptLine.startTimestampMs, transcriptLine.endTimestampMs);
+
+    const template = document.getElementById(TRANSCRIPT_TEMPLATE);
+    this.attachShadow({mode: 'open'});
+    this.shadowRoot.appendChild(template.content.cloneNode(true));
+    this.updateTemplateSlot(TRANSCRIPT_SLOT_TIME_RANGE, timestampRange);
+    this.updateTemplateSlot(TRANSCRIPT_SLOT_CONTENT, transcriptLine.content);
+
+    this.startTimestampMs = transcriptLine.startTimestampMs;
+    this.endTimestampMs = transcriptLine.endTimestampMs;
```

Comment:
It might be better to just save everything with `this.transcriptLine = transcriptLine` instead of copying over the various values you'll need later.